astromodels/functions/functions.py

```diff
@@ -560,6 +560,136 @@ def from_unit_cube(self, x):
 
         return res
 
+# noinspection PyPep8Naming
+class TruncatedGaussian(Function1D):
+    r"""
+    description :
+
+        A  truncated Gaussian function defined on the interval between the lower_bound (a) and upper_bound (b)
+
+    latex : $\begin{split}f(x;\mu,\sigma,a,b)=\frac{\frac{1}{\sigma} \phi\left( \frac{x-\mu}{\sigma} \right)}{\Phi\left( \frac{b-\mu}{\sigma} \right) - \Phi\left( \frac{a-\mu}{\sigma} \right)}\\\phi\left(z\right)=\frac{1}{\sqrt{2 \pi}}\exp\left(-\frac{1}{2}z^2\right)\\\Phi\left(z\right)=\frac{1}{2}\left(1+erf\left(\frac{z}{\sqrt(2)}\right)\right)\end{split}$
+
+    parameters :
+
+        F :
+
+            desc : Integral between -inf and +inf. Fix this to 1 to obtain a Normal distribution
+            initial value : 1
+
+        mu :
+
+            desc : Central value
+            initial value : 0.0
+
+        sigma :
+
+            desc : standard deviation
+            initial value : 1.0
+            min : 1e-12
+
+        lower_bound :
+
+            desc: lower bound of gaussian, setting to -np.inf results in half normal distribution
+            initial value : -1.
+
+        upper_bound :
+
+            desc: upper bound of gaussian  setting to np.inf results in half normal distribution
+            initial value : 1.
+
+
+    tests :
+        - { x : 0.0, function value: 0.3989422804014327, tolerance: 1e-10}
+        - { x : -1.0, function value: 0.24197072451914337, tolerance: 1e-9}
+
+    """
+
+    __metaclass__ = FunctionMeta
+
+    # Place this here to avoid recomputing it all the time
+
+    __norm_const = 1.0 / (math.sqrt(2 * np.pi))
+
+    def _set_units(self, x_unit, y_unit):
+
+        # The normalization is the integral from -inf to +inf, i.e., has dimensions of
+        # y_unit * x_unit
+        self.F.unit = y_unit * x_unit
+
+        # The mu has the same dimensions as the x
+        self.mu.unit = x_unit
+
+        # The lower_bound has the same dimensions as the x
+        self.lower_bound.unit = x_unit
+
+        # The upper_bound has the same dimensions as the x
+        self.upper_bound.unit = x_unit
+
+        # sigma has the same dimensions as x
+        self.sigma.unit = x_unit
+
+
+
+
+    # noinspection PyPep8Naming
+    def evaluate(self, x, F, mu, sigma, lower_bound, upper_bound):
```

Comment:
thanks for this addition!

Is this faster than using scipy.stats.truncnorm?

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.truncnorm.html